xbmc/storage/discs/AbstractDiscDriveHandler.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "AbstractDiscDriveHandler.h"
+
+#include <stdexcept>
+
+DriveState AbstractDiscDriveHandler::GetDriveState(const std::string& devicePath)
+{
+  std::logic_error("GetDriveState: Method not allowed in abstract class");
```

Comment:
as you declared methods as pure virtual, there should be no need to define those methods here, no?